internal/cluster/distribution/pke/pkeaws/pkeawsadapter/image_selector.go

```diff
@@ -84,5 +84,18 @@ func (s CloudinfoImageSelector) SelectImage(ctx context.Context, criteria pkeaws
 		)
 	}
 
+	// As a result of a bug in cloudinfo,
+	// the returned item might be empty
+	// See https://github.com/banzaicloud/cloudinfo/pull/356
```

Comment:
I wouldn't handle this case here, at least remove this check once we are sure that the bug is fixed.

Reply:
That's the plan, but I didn't have much of a choice if I wanted to let this PR in.